src/hotspot/os/linux/os_linux.cpp

```diff
@@ -816,6 +819,32 @@ static size_t get_static_tls_area_size(const pthread_attr_t *attr) {
   return tls_size;
 }
 
+// In glibc versions prior to 2.27 the guard size mechanism
+// was not implemented properly. The posix standard requires adding
+// the size of the guard pages to the stack size, instead glibc
+// took the space out of 'stacksize'. Thus we need to adapt the requested
+// stack_size by the size of the guard pages to mimic proper behaviour.
+// The fix in glibc 2.27 has now been backported to numerous earlier
+// glibc versions so we need to do a dynamic runtime check.
+bool os::Linux::AdjustStackSizeForGuardPages = true;
+
+static void init_adjust_stack_for_guard_pages() {
+  get_minstack_init();
+  if (_get_minstack_func != nullptr) {
+    pthread_attr_t attr;
+    pthread_attr_init(&attr);
+    size_t min_stack = _get_minstack_func(&attr);
+    pthread_attr_setguardsize(&attr, 16*K );
+    size_t min_stack2 = _get_minstack_func(&attr);
+    pthread_attr_destroy(&attr);
+    // If the minimum stack size changed when we added the guard pages
+    // then we need to perform the adjustment.
+    os::Linux::AdjustStackSizeForGuardPages = (min_stack2 - min_stack > 0);
+    log_info(os)("- glibc stack size guard page adjustment is %sneeded",
```

Comment:
Why the leading dash?

Reply:
It was to make it look like part of the version check:
```
[0.003s][info ][os] HotSpot is running with glibc 2.17, NPTL 2.17
[0.003s][info ][os] - glibc stack size guard page adjustment is not needed
```
but I realize now that can be messed up by other logging:
```
[0.003s][info ][os] HotSpot is running with glibc 2.17, NPTL 2.17
[0.003s][info ][os,thread] Lookup of __pthread_get_minstack succeeded
[0.003s][info ][os       ] - glibc stack size guard page adjustment is not needed
```
so I will change it.